test/e2e/storage/vsphere_volume_vsan_policy.go

```diff
@@ -181,21 +194,72 @@ var _ = framework.KubeDescribe("VSAN policy support for dynamic provisioning [Vo
 		scParameters[Policy_ObjectSpaceReservation] = ObjectSpaceReservationCapabilityVal
 		scParameters[Datastore] = VmfsDatastore
 		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
-		err := invokeInvalidVSANPolicyTestNeg(client, namespace, scParameters)
+		err := invokeInvalidPolicyTestNeg(client, namespace, scParameters)
 		// This will make sure err is always returned.
 		Expect(err).To(HaveOccurred())
 		errorMsg := "The specified datastore: \\\"" + VmfsDatastore + "\\\" is not a VSAN datastore. " +
 			"The policy parameters will work only with VSAN Datastore."
-		framework.Logf("errorMsg: %+q", errorMsg)
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	It("verify an existing and compatible SPBM policy is honored for dynamically provisioned pvc using storageclass", func() {
+		By(fmt.Sprintf("Invoking Test for SPBM policy: %s", os.Getenv("VSPHERE_SPBM_GOLD_POLICY")))
+		scParameters[SpbmStoragePolicy] = os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
+		scParameters[DiskFormat] = "thin"
+		framework.Logf("Invoking Test for SPBM storage policy: %+v", scParameters)
+		invokeValidPolicyTest(f, client, namespace, scParameters)
+	})
+
+	It("verify if a SPBM policy is not honored on a non-compatible datastore for dynamically provisioned pvc using storageclass", func() {
+		By(fmt.Sprintf("Invoking Test for SPBM policy: %s and datastore: %s", os.Getenv("VSPHERE_SPBM_TAG_POLICY"), VsanDatastore))
+		scParameters[SpbmStoragePolicy] = os.Getenv("VSPHERE_SPBM_TAG_POLICY")
+		scParameters[Datastore] = VsanDatastore
+		scParameters[DiskFormat] = ThinDisk
+		framework.Logf("Invoking Test for SPBM storage policy on a non-compatible datastore: %+v", scParameters)
+		err := invokeInvalidPolicyTestNeg(client, namespace, scParameters)
+		// This will make sure err is always returned.
+		Expect(err).To(HaveOccurred())
+		errorMsg := "User specified datastore: \\\"" + VsanDatastore + "\\\" is not compatible with the storagePolicy: \\\"" + os.Getenv("VSPHERE_SPBM_TAG_POLICY") + "\\\""
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
An error *did* occur but not the error you were expecting. In this case you should add the extra string parm, shown above, to include what type of error you were expecting.